src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -407,18 +408,111 @@ private void LayoutChildren(Span<DoubleRange> calculatedPixelColumns, Span<Doubl
 				}
 			}
 
+			// Auto columns should take up the larger of their measured width and their MinWidth
+			for (int i = 0; i < columns.Length; i++)
+			{
+				if (columns[i].Width.IsAuto)
+				{
+					calculatedPixelColumns.Span[i] = Math.Max(calculatedPixelColumns.Span[i], columns[i].MinWidth);
+				}
+			}
+
 			// Star size: We always measure to set the desired size, but measuring would only be required for the columns calculation when the Star doesn't mean remaining space.
 			var usedWidth = calculatedPixelColumns.Span.Sum();
-			double remainingWidth = Math.Max(0, availaibleWidth - usedWidth);
-			var totalStarSizedWidth = GetTotalStarSizedWidth(columns);
-			var defaultStarWidth = remainingWidth / totalStarSizedWidth;
-
-			var starCalculatedPixelColumns = columns
-				.SelectToMemory(
-					(c, i) => c.Width.IsStarSize ?
-						c.Width.StarSize.GetValueOrDefault() * defaultStarWidth :
-						calculatedPixelColumns.Span[i]
-				);
+			var initialRemainingWidth = Math.Max(0, availaibleWidth - usedWidth);
+			var initialTotalStarSizedWidth = GetTotalStarSizedWidth(columns);
+
+			var remainingWidth = initialRemainingWidth;
+			var totalStarSizedWidth = initialTotalStarSizedWidth;
+			var unitStarWidth = remainingWidth / totalStarSizedWidth;
+
+			var maxTries = starSizeChildren.Span.Length * 2 + 1;
```

Comment:
What's the idea behind maxTries being star children count * 2 + 1?

Reply:
+1 to always run at least once, star children count because it shouldn't iterate more times than there are star children, *2 for margin of error.

Note that in the case there are no MinWidths or MaxWidths set it will only run once and will behave as before.